library/std/src/net/ip.rs

```diff
@@ -1119,16 +1115,16 @@ impl Ipv6Addr {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn segments(&self) -> [u16; 8] {
-        let arr = &self.inner.s6_addr;
+        let [a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p] = self.inner.s6_addr;
```

Comment:
Hmm, this one with 16 locals seems a bit much. The correct even-odd pairs are pretty easy to see with indexing, but not as obvious with letters - only that they're sequential.

Codegen looks the same either way: https://rust.godbolt.org/z/7dWca9

Reply:
I thought about this too but it would align more to the rest I think, I would probably lost count of the numbers for alphabet starting from g. Maybe it could be?
```rust
         let [a0, a1, b0, b1, c0, c1, d0, d1, e0, e1, f0, f1, g0, g1, h0, h1] = self.inner.s6_addr;
         [
             u16::from_be_bytes([a0, a1]),
             u16::from_be_bytes([b0, b1]),
             u16::from_be_bytes([c0, c1]),
             u16::from_be_bytes([d0, d1]),
             u16::from_be_bytes([e0, e1]),
             u16::from_be_bytes([f0, f1]),
             u16::from_be_bytes([g0, g1]),
             u16::from_be_bytes([h0, h1]),
         ]

```

Reply:
I try to casting union fields in #75085

Reply:
Wow, I didn't know this works. Looks like a hack to me.